### PR TITLE
lightworks: 14.0.0 -> 2021.2

### LINKED ISSUE #80535: Lightworks are missing libraries
**Describe the bug**
I cannot run lightworks from 19.09 branch. Running `lightworks-14.0.0` produces `/nix/store/qpip8xrj0ghkrkkw86ljl4mz5rw6fprl-lightworks-14.0.0/lib/lightworks/ntcardvt: error while loading shared libraries: libGL.so.1: cannot open shared object file: No such file or directory`. I am using ThinkPad W530 with i7-3740QM. I use the integrated Intel GPU and optionally use optirun to run tasks on the NVIDIA QUADRO 1000M. When I run `optirun lightworks-14.0.0` I get `/nix/store/qpip8xrj0ghkrkkw86ljl4mz5rw6fprl-lightworks-14.0.0/lib/lightworks/ntcardvt: error while loading shared libraries: libjpeg.so.8: cannot open shared object file: No such file or directory`.

**To Reproduce**
Steps to reproduce the behavior without optirun:
1. `nix-shell -p lightworks`
2. `lightworks-14.0.0`

Steps to reproduce the behavior with optirun:
1. Add `hardware.bumblebee.enable = true;` to your `configuration.nix`
2. `nix-shell -p lightworks`
3. `optirun lightworks-14.0.0`

**Expected behavior**
Lightwork would start up.

**Metadata**
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.101, NixOS, 19.09.2036.c49da6435f3 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(vojta001): `""`
 - channels(root): `"nixos-19.09.2036.c49da6435f3"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by vojta001**
Fixes #80535. Supersedes #83702.

###### Motivation for this change
The old version is broken in nixpkgs, because libjpeg got updated in the meantime.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vojta001** [comment]
Current maintainer: @antonxy

File: pkgs/applications/video/lightworks/default.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { lib, stdenv, fetchurl, dpkg, makeWrapper, buildFHSUserEnv
 , gtk3, gdk-pixbuf, cairo, libjpeg_original, glib, pango, libGLU
-, nvidia_cg_toolkit, zlib, openssl, portaudio
+, libGL, nvidia_cg_toolkit, zlib, openssl, libuuid , alsaLib, libudev
```
Comment (OPNA2608) [inline pkgs/applications/video/lightworks/default.nix:3]:
```suggestion
, libGL, nvidia_cg_toolkit, zlib, openssl, libuuid , alsaLib, udev
```

File: pkgs/applications/video/lightworks/default.nix [hunk 2]
```diff
@@ -11,22 +11,25 @@ let
...
     pango
+    libGL
     libGLU
     nvidia_cg_toolkit
     zlib
     openssl
-    portaudio
+    libuuid
+    alsaLib
+    libudev
```
Comment (OPNA2608) [inline pkgs/applications/video/lightworks/default.nix:21]:
```suggestion
    udev
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
`libudev` is an alias for `udev`.

Failed alias check: https://gist.github.com/GrahamcOfBorg/fd7cb1dd0bebd9952131ae3208199891

> anonymous function at <…>/pkgs/applications/video/lightworks/default.nix:1:1 called without required argument 'libudev'

**Comment by vojta001** [comment]
@OPNA2608 Thanks! Fixed

**Comment by OPNA2608** [comment]
Result of `nixpkgs-review pr 121663` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lightworks</li>
  </ul>
</details>

Binary launches, that's all I've tested though.

**Comment by OPNA2608** [comment]
Seems like there's 2021.2 now, can we bump to that? I'll make sure to find someone with merge permissions afterwards.

File: pkgs/applications/video/lightworks/default.nix [hunk 3]
```diff
@@ -11,22 +11,25 @@ let
...
-    version = "14.0.0";
+    version = "2021.2";
     pname = "lightworks";
 
     src =
       if stdenv.hostPlatform.system == "x86_64-linux" then
         fetchurl {
-          url = "http://downloads.lwks.com/v14/lwks-14.0.0-amd64.deb";
-          sha256 = "66eb9f9678d979db76199f1c99a71df0ddc017bb47dfda976b508849ab305033";
+          url = "https://cdn.lwks.com/releases/2021.2/lightworks_2021.2_r128258.deb";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/lightworks/default.nix:31]:
```suggestion
          url = "https://cdn.lwks.com/releases/${version}/lightworks_${version}_r128258.deb";
```

the rXXXX part could be replaced with an update script.

**Comment by OPNA2608** [comment]
> Seems like there's 2021.2 now, can we bump to that? I'll make sure to find someone with merge permissions afterwards.

Oh, it's already bumping to that version. Can you fix the commit & PR titles please? They refer to 2021.1.

**Comment by vojta001** [comment]
@OPNA2608 Fixed the confusing title and commit message.

File: pkgs/applications/video/lightworks/default.nix (hunk 3 above)
Comment (vojta001) [reply pkgs/applications/video/lightworks/default.nix:31]: I am sorry, I don't fully understand. Could you please elaborate?

**Comment by OPNA2608** [comment]
Seems like upstream pushed a new version 2021.2.1 (subversion only exists on their download server, download page still refers to it as 2021.2 but links to this new deb):

[Lightworks 2021.2 Installer Update Revision 128456](https://www.lwks.com/index.php?option=com_kunena&func=view&catid=19&id=239738&Itemid=81)
https://cdn.lwks.com/releases/2021.2.1/lightworks_2021.2_r128456.deb

**Comment by vojta001** [comment]
@OPNA2608 Thanks for notifying. Updated.

File: pkgs/applications/video/lightworks/default.nix [hunk 4]
```diff
@@ -11,22 +11,25 @@ let
...
-    version = "14.0.0";
+    version = "2021.2.1";
     pname = "lightworks";
 
     src =
       if stdenv.hostPlatform.system == "x86_64-linux" then
         fetchurl {
-          url = "http://downloads.lwks.com/v14/lwks-14.0.0-amd64.deb";
-          sha256 = "66eb9f9678d979db76199f1c99a71df0ddc017bb47dfda976b508849ab305033";
+          url = "https://cdn.lwks.com/releases/2021.2.1/lightworks_2021.2_r128456.deb";
```
Comment (OPNA2608) [inline pkgs/applications/video/lightworks/default.nix:31]:
```suggestion
          url = "https://cdn.lwks.com/releases/${version}/lightworks_${lib.versions.majorMinor version}_r128456.deb";
```

You could also move the `128456` out into a `rev = "128456";` so for the next bump the URL wouldn't need to be changed (`…/lightworks_${lib.versions.majorMinor version}_r${rev}.deb`), just version, rev & sha256.

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/lightworks/default.nix (hunk 4 above)
Comment (vojta001) [reply pkgs/applications/video/lightworks/default.nix]: Done

**Review by OPNA2608 — APPROVED** [review]
LGTM. @SuperSandro2000 what do you think?

**Comment by vojta001** [comment]
Thanks. What's you opinion on backporting this to 21.05?

**Comment by OPNA2608** [comment]
I don't see any problem with backporting it to 21.05.

**Pull request merged**